javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -0,0 +1,109 @@
+package java.util.regex
+
+import scalanative.native._, stdlib._, stdio._
+import scalanative.runtime.struct
+
+@extern
+object Regex {
```

Comment:
For extern objects, we've been using lowercase name that corresponds to the name of the header so far. Lets keep this convention.